.travis.yml

```diff
@@ -31,60 +24,60 @@ matrix:
         - DEPS=lowest
     - php: 5.6
       env:
-        - DEPS=latest
+        - DEPS=locked
         - TEST_COVERAGE=true
-        - DEPLOY_DOCS="$(if [[ $TRAVIS_BRANCH == 'master' && $TRAVIS_PULL_REQUEST == 'false' ]]; then echo -n 'true' ; else echo -n 'false' ; fi)"
-        - PATH="$HOME/.local/bin:$PATH"
+    - php: 5.6
+      env:
+        - DEPS=latest
     - php: 7
       env:
         - DEPS=lowest
     - php: 7
       env:
-        - DEPS=latest
+        - DEPS=locked
         - CS_CHECK=true
+    - php: 7
+      env:
+        - DEPS=latest
     - php: 7.1
       env:
         - DEPS=lowest
     - php: 7.1
       env:
+        - DEPS=locked
+    - php: 7.1
+      env:
         - DEPS=latest
-    - php: hhvm
+    - php: 7.2
       env:
         - DEPS=lowest
-    - php: hhvm
+    - php: 7.2
+      env:
+        - DEPS=locked
+    - php: 7.2
       env:
         - DEPS=latest
   allow_failures:
-    - php: hhvm
+    - php: 7.2
 
 before_install:
-  - if [[ $TRAVIS_PHP_VERSION != "hhvm" && $TEST_COVERAGE != 'true' ]]; then phpenv config-rm xdebug.ini ; fi
+  - if [[ $TEST_COVERAGE != 'true' ]]; then phpenv config-rm xdebug.ini || return 0 ; fi
   - travis_retry composer self-update
 
 install:
-  - travis_retry composer install $COMPOSER_ARGS
+  - travis_retry composer install $COMPOSER_ARGS --ignore-platform-reqs
```

Comment:
@webimpress — No - use PHP 7 or 7.1 for the `composer.lock` generation; otherwise, we're using a lower set of dependencies in many cases due to the fact that some libraries and/or their versions won't work with PHP 5 (e.g., PHPUnit 6). Additionally, we have 2, and shortly to be 3, versions of PHP 7 we test against, but only 1 version of PHP 5 we test against. As such, use the versions supported by the majority of PHP versions we test, not the other way around.

@kokspflanze — we only use `--ignore-platform-reqs` for the initial install from the lockfile; all other operations take it into account. The line I shared earlier for installing from `$LEGACY_DEPS` is tested and works well, and ensures we have a consistent installation across that particular PHP version, and that the dependencies _work_ for that PHP version. Further, if we add any additional dependencies that depend on PHP 7+, we'll know about them due to Travis erroring on that particular job, and we can then evaluate whether or not we need to change any of our constraints.